experimental/packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -57,7 +64,22 @@ export class NodeSDK {
    * Create a new NodeJS SDK instance
    */
   public constructor(configuration: Partial<NodeSDKConfiguration> = {}) {
-    this._resource = configuration.resource ?? new Resource({});
+    const resource = configuration.resource ?? new Resource({});
+
+    const { resourceAttributes } = configuration;
+
+    this._resource = resourceAttributes === undefined
+    ? resource
+    : resource.merge(new Resource(
+      Object.keys(resourceAttributes).reduce((attributes, key) => {
+        attributes[
+          SemanticResourceAttributes[
```

Comment:
I think this doesn't work if a user passes a resouce name which is not part of `@opentelemetry/semantic-conventions`.
As far as I know users are free to add resource attributes not actually specified.

Reply:
Why would we not just use the strings directly provided by the user? If they want to use the semconv package nobody is stopping them from using it.

Reply:
Ok will revisit allowing for free form string

Reply:
If we accept any resource key then this code can be rewritten as:
`resource.merge(new Resource(resourceAttributes))`.

This means that this feature only saves the user from writing `myResource.merge(new Resource(...))` in his code.

I think that adding just a single config option to set `serviceName` can be more straightforward for users to understand and use. This resource attribute is the only required one in the spec. It will also mean that users will not have to discover and use the relevant attribute key name ("service.name"), or import semantic-conventions package in their code.

For more advanced use cases (other attributes), I am in favor of just letting users define and merge the resource themselves.

Reply:
Agreed, I'll go back to original plan and just add a serviceName prop